zipline/algorithm.py

```diff
@@ -860,18 +968,48 @@ def record(self, *args, **kwargs):
             self._recorded_vars[name] = value
 
     @api_method
-    def set_benchmark(self, benchmark_sid):
+    def set_benchmark(self, benchmark):
+        """Set the benchmark asset.
+
+        Parameters
+        ----------
+        benchmark : Asset
+            The asset to set as the new benchmark.
+
+        Notes
+        -----
+        Any dividends payed out for that new benchmark asset will be
+        automatically reinvested.
+        """
         if self.initialized:
             raise SetBenchmarkOutsideInitialize()
 
-        self.benchmark_sid = benchmark_sid
+        self.benchmark_sid = benchmark
```

Comment:
I would want to do that in a different change set since it would have downstream affects